src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -373,7 +371,33 @@ private function isMaxDepthReached(array $attributesMetadata, $class, $attribute
     }
 
     /**
-     * Gets the cache key to use.
+     * Overwritten to update the cache key for the child.
+     *
+     * We must not mix up the attribute cache between parent and children.
+     *
+     * {@inheritdoc}
+     */
+    protected function createChildContext(array $parentContext, $attribute/*, string $format = null*/)
+    {
+        if (\func_num_args() >= 3) {
+            $format = \func_get_arg(2);
+        } else {
+            @trigger_error(sprintf('Method %s::%s() will have a 3rd `string $format = null` argument in version 5.0. Not defining it is deprecated since Symfony 3.4.', \get_class($this), __FUNCTION__), E_USER_DEPRECATED);
```

Comment:
@dunglas is this correct like that? is it ok to deprecate in version 3 for something that will only break in version 5?

Reply:
it's not, this should be done on master instead

Reply:
makes sense. i removed the trigger_error.

ping me when this has been merged to master, then i add it back there both here and for the parent method.